trk/src/Screens/Home_Backend/HomeScreenLogic.js

```diff
@@ -21,6 +26,22 @@ export const useHomeScreenLogic = (props) => {
   // user check
   const { currentUser } = useContext(AuthContext);
 
+  const dispatch = useDispatch();
+
+
+  const checkConnectivity = (climbId) => {
+    NetInfo.fetch().then(state => {
+      console.log("Is connected?", state.isConnected);
+      if (state.isConnected) {
+        console.log('Navigating');
+        navigation.navigate('Detail', { climbId: climbId[0], isFromHome: true });
+      } else {
+        dispatch(addClimb(climbId[0], currentUser));
+        Alert.alert('Offline Action', 'Your action is saved and will be processed when you\'re online.', [{ text: 'OK' }]);
```

Comment:
Is the user still redirected to ClimbDetail upon read? (no matter whether it's successful or not).